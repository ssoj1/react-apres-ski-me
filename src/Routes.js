import React from "react";
import { Switch, Route } from "react-router-dom";
import Home from "./Home";
import Menu from "./Menu";
import ItemDetailCard from "./ItemDetailCard";
import AddItemForm from "./AddItemForm";


/** Site-wide routes.
 *
 * Visiting a non-existent snack or drink reroutes to the /snacks or /drinks
 * page, respectively. 
 * Visiting a non-existent route redirects to the homepage.
 * 
 * Props: 
 * - snacks - array of objects like: [{snack}, ...]
 * - drinks - array of objects like: [{drink}, ...]
 * - addItem - a function run on the parent that adds an item to the API and 
 * snacks or drinks array, depending on type
 * 
 * State: 
 * - none
 * 
 * App -> Routes -> { Home, Menu, ItemDetailCard, AddItemForm }
 */

function Routes({ snacks, drinks, addItem }) {
  console.debug("* Routes", { snacks, drinks, addItem });

  return (
    <div className="App">
      <main>
        <Switch>
          <Route exact path="/">
            <Home snacks={snacks} drinks={drinks} />
              </Route>

              <Route exact path="/snacks">
                <Menu items={snacks} type="Snacks" />
              </Route>

              <Route exact path="/snacks/:id">
                <ItemDetailCard items={snacks} cantFind="/snacks" />
              </Route>

              <Route exact path="/drinks">
                <Menu items={drinks} type="Drinks" />
              </Route>

              <Route exact path="/drinks/:id">
                <ItemDetailCard items={drinks} cantFind="/drinks" />
              </Route>

              <Route exact path="/additem">
                <AddItemForm addItem={addItem} />
              </Route>

              <Route>
                <p>
                Hmmm.I can't seem to find that page.
                </p>
              </Route>
        </Switch>
      </main>
    </div>
    );
}

export default Routes;
