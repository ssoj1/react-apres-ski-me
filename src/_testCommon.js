
// sample data to be used in testing

const snacks = [
    {
        "id": "test-snack",
        "name": "Test Snack",
        "description": "Test snack description.",
        "recipe": "Test snack recipe.",
        "serve": "Test snack serve."
      }
];

const drinks = [
    {
        "id": "test-drink",
        "name": "Test Drink",
        "description": "Test drink description.",
        "recipe": "Test drink recipe.",
        "serve": "Test drink serve."
      }
]; 

export { snacks, drinks };