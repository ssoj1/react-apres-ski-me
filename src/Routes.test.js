import React from "react";
import { render } from "@testing-library/react";
import Routes from "./Routes";
import { MemoryRouter } from "react-router";
import { snacks, drinks } from "./_testCommon";

  

test("renders without crashing", function () {
  render(
      <MemoryRouter>
          <Routes snacks={snacks} drinks={drinks} />
      </MemoryRouter>,
  );
});

test("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
          <Routes snacks={snacks} drinks={drinks} />
      </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});