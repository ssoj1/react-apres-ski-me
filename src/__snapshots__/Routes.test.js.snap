// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <main>
      <section
        class="col-md-8"
      >
        <div
          class="card"
        >
          <div
            class="text-center card-body"
          >
            <div
              class="card-title"
            >
              <h3
                class="font-weight-bold"
              >
                Welcome to Tahoe's premier dive cafe!
              </h3>
              <p>
                We have 1 snacks and 1 drinks to choose from.
              </p>
            </div>
          </div>
        </div>
      </section>
    </main>
  </div>
</DocumentFragment>
`;
